src/components/form/file-upload/file-upload.tsx

```diff
@@ -0,0 +1,177 @@
+import { FC, DragEvent, useState, useRef, useId, ChangeEvent } from 'react';
+import { FileUploadProps } from './types';
+import { LabelSize } from '../../typography/label/types';
+import { Label } from '../../typography/label/label';
+import clsx from 'clsx';
+import { IconSize, IconUpload } from '../../../elements/icons';
+import { Button } from '../../button/button';
+import { ButtonSize } from '../../button/types';
+import { Variant } from '../../../utlis/types';
+
+export const FileUpload: FC<FileUploadProps> = ({
+    label = '',
+    labelFileSize = '',
+    labelButton = '',
+    Icon,
+    IconActionLeft,
+    IconActionRight,
+    onLoadFile,
+}) => {
+    const fileInputId = useId();
+    const inputReference = useRef<HTMLInputElement>(null);
+    const [dragIsOver, setDragIsOver] = useState(false);
+    const [isValidFileType, setIsValidFileType] = useState(true);
+    const [isValidFileSize, setIsValidFileSize] = useState(true);
```

Comment:
Maybe create a context for state handling instead of 3 useStates (example in like-button)?